src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -368,6 +369,13 @@ public void before() throws Throwable {
 
         try {
             jenkins = hudson = newHudson();
+            // If the initialization graph is corrupted, we cannot expect that Jenkins is in the good shape.
+            // Likely it is an issue in @Initializer() definitions (see JENKINS-37759).
+            // So we just fail the test.
+            if (jenkins.getInitLevel() != InitMilestone.COMPLETED) {
+                throw new Exception("Jenkins initialization has not reached the COMPLETED initialization stage. Current state is " + jenkins.getInitLevel() +
```

Comment:
Maybe. I still want it to pass through the Exception handler, hence I decided to use common Exception